tests/res/enkf/test_row_scaling_case.py

```diff
@@ -458,3 +463,57 @@ def test_2ministep(self):
 
                     diff = abs(v1 - v2)
                     assert diff < 0.05
```

Comment:
This fails on python 3.9, why is it set to 0.05?

Reply:
The problem here is that there is a stochastic element and the test is just "somewhat close" - 0.05 was OK on my machine. I realize this is an extremely weak test, but to actually get control over the random state to level needed to get this identical is a quite large job.

I have increased the test ensemble size which should reduce fluctuations but of course also increase runtime; alternatively the actual numerical test can just be removed?

Reply:
I am worried that this flakyness will hit us at later point in time. Maybe it is best to remove it?

Reply:
I don't agree with the term flakyness - the situation is understood; but I agree that the value is very limited - happy to remove it.

Reply:
It will not occasionally fail? or is it deterministic and will always fail on python 3.9? What does the it mean then; that the difference between one ministep and two ministeps is less than 5%?

Reply:
No it will not occasionally fail; by setting the seed in the test the sampling is deterministic and will be reproducible. The difference comes from the function `obs_data_allocE`. Assume that we have a reproducible stream of random numbers:
```
R = {R_1, R_2, R_3, R_4, R_5, R_6}
``` 

then in the case of one ministep with two observations and three ensemble members we will get:
```
     [ R_1  R_2 ]
E =  [ R_3  R_4 ]
     [ R_5  R_6 ]
``` 
whereas we with two ministeps with one observation in each we get the two E matrices:
```
     [R_1]           [R_4] 
E1 = [R_2],     E2 = [R_5]     
     [R_3]           [R_6] 
```

i.e. because the the ensemble members and observations are interleaved in the E matrix 2 x 1 != 2.

I must admit I was little surprised by the Python 3.9 failure - my guess is/was that there has been some change in the Python random number functionality so we do not get the same sequence on 3.9 as on 3.8 - I  have not investigated that further.

> What does the it mean then; that the difference between one ministep and two ministeps is less than 5%?

That they are *quite close*; which is what we should expect. It should be added that the difference is not 5% - the comparison is absolute.

Reply:
Would it be possible to attach a mocked random number generator to the enkf_main object before creating E wich would then create 
```
     [R_1]           [R_4] 
E1 = [R_2],     E2 = [R_5]     
     [R_3]           [R_6] 

     [ R_1  R_4 ]
E =  [ R_2  R_5 ]
     [ R_3  R_6 ]

```

Reply:
>Would it be possible to attach a mocked random number generator to the enkf_main object before creating E which would then create?

Everything is possible - but I would say: *No it is not worth it*. What you suggest would certainly be a necessary step - I have not verified if it whether it is sufficient, my guess is that additional changes would be necessary.

Reply:
Then I would say the numeric comparison is not having any value, and it can be removed. I guess the test only shows that it is possible to split up in multiple ministeps, but it is not identical to doing it in one step.

Reply:
>  [...] and it can be removed

I agree. I have removed it - ans reset the test ensemble size to 10.